src/megaapi_impl.cpp

```diff
@@ -20674,27 +20979,181 @@ void MegaApiImpl::sendPendingRequests()
+                syncError = ACCOUNT_BLOCKED;
+            }
+
+            bool enabled = request->getFlag();
+            bool temporaryDisabled = request->getNumDetails();
+
+            if (!enabled && temporaryDisabled)
+            {
+                if (!syncError) syncError = UNKNOWN_TEMPORARY_ERROR;
+                enabled = true; //we consider enabled when it is temporary disabled
+            }
+
+            SyncConfig syncConfig{nextSyncTag, localPath, request->getNodeHandle(),
+                                  static_cast<fsfp_t>(request->getNumber()),
+                                  regExpToVector(request->getRegExp()), enabled};
+
+            if (temporaryDisabled)
+            {
+                if (!syncError)
+                {
+
+                }
+                syncConfig.setError(syncError);
+            }
+
+            if (client->syncConfigs)
+            {
+                client->syncConfigs->insert(syncConfig);
             }
             else
             {
-                delete sync;
+                e = API_ENOENT;
+            }
+
+            if (!e)
+            {
+                request->setNumber(nextSyncTag);
+                fireOnRequestFinish(request, MegaError(API_OK));
+            }
+            break;
+        }
+        case MegaRequest::TYPE_COPY_CACHED_STATUS:
+        {
+            auto number = request->getNumber();
+            int businessStatusValue = static_cast<int>(number / 1000000);
+            number = number % 1000000;
+            int blockedStatusValue =  static_cast<int>(number / 1000);
+            int storageStatusValue = number % 1000;
+
+
+            using CS = CacheableStatus;
+            using CType = CacheableStatus::Type;
+
+            auto loadAndPersist = [this](CType type, int value) -> bool
+            {
+                if (value == 999)
+                {
+                    return false;
+                }
+                auto status = client->mCachedStatus[type];
+                if (status)
+                {
+                    return false;
+                }
+                status = std::make_shared<CS>(type, value);
+
+                client->loadCacheableStatus(status);
+
+                if (client->sctable)
+                {
+                    LOG_verbose << "Adding/updating migrated status to database: " << status->type() << " = " << status->value();
+                    if (!(client->sctable->put(MegaClient::CACHEDSTATUS, status.get(), &client->key)))
+                    {
+                        LOG_err << "Failed to add/update migrated status to db: "
+                                    << status->type() << " = " << status->value();
+                    }
+                }
+            };
+
+            loadAndPersist(CType::STATUS_STORAGE, storageStatusValue);
+            loadAndPersist(CType::STATUS_BUSINESS, businessStatusValue);
+            loadAndPersist(CType::STATUS_BLOCKED, blockedStatusValue);
+
+            if (!e)
```

Comment:
I think you need to set `e` to the return value of each call to `loadAndPersist()`. Otherwise, this `if` is useless.